.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+Contributing
+============
+
+Thanks for taking the time to contribute to Orange!
+
+Please submit contributions in accordance with the flow explained in the
+[GitHub Guides].
+
+[GitHub Guides]: https://guides.github.com/
+
+
+Reporting bugs
+--------------
+Please report bugs according to established [bug reporting guidelines].
+At least, include a method to reproduce the bug (if consistently
+reproducible) and a screenshot (if applicable).
+
+[bug reporting guidelines]: https://www.google.com/search?q=reporting+bugs
+
+
+Coding style
+------------
+
+Roughly conform to [PEP-8] style guide for Python code. Whenever PEP-8 is
+undefined, adhere to [Google Python Style Guide].
+
+In addition, we add the following guidelines:
+
+* Only ever `import *` to make objects available in another namespace,
+  preferably in *\_\_init\_\_.py*. Everywhere else use explicit object
+  imports.
+* Use [Napoleon]-comaptible (e.g. NumPy style) docstrings, preferably with
+  [tests].
+* When instantiating Qt widgets, pass static property values as
+  [keyword args to the constructor] instead of calling separate property
+  setters later. For example, do:
+
+      view = QListView(alternatingRowColors=True,
+                       selectionMode=QAbstractItemView.ExtendedSelection)
+
+  instead of:
+
+      view = QListView()
+      view.setAlternatingRowColors(True)
+      view.setSelectionMode(QAbstractItemView.ExtendedSelection)
+
+* Each Orange widget module, or better still, each Python module (within
+  reason) should have a `__name__ == '__main__'`-fenced code block that
+  shows/tests the gist of that module in a user-friendly way.
+
+[PEP-8]: https://www.python.org/dev/peps/pep-0008/
+[Google Python Style Guide]: https://google.github.io/styleguide/pyguide.html
+[Napoleon]: http://www.sphinx-doc.org/en/stable/ext/napoleon.html
+[keyword args to the constructor]: http://pyqt.sourceforge.net/Docs/PyQt5/qt_properties.html
+
+
+Tests
+-----
+[tests]: #tests
+Ensure the tests pass by running:
+
+    python setup.py test
+
+If you contribute new code, make [unit tests] for it in _Orange/tests_.
+
+Prefer [doctests] for public APIs. Note, we unit-test doctests with
+`NORMALIZE_WHITESPACE` and `ELLIPSIS` options enabled, so you can use them
+implicitly.
+
+[unit tests]: https://en.wikipedia.org/wiki/Unit_testing
+[doctests]: https://en.wikipedia.org/wiki/Doctest
+
+
+Commit messages
+---------------
+Make a separate commit for each logical change you introduce. We prefer
+short commit messages with descriptive titles. For a general format see
+[Commit Guidelines]. E.g.:
+
+> io: Fix reader for XYZ file format
```

Comment:
This example starts with a tag, but the instructions don't mention them (enough).
I think we want to emphasize this more.